### PR TITLE
simplified PoolBagEntry creation, removed PoolElf#resetPoolEntry()

### PR DESCRIPTION
**Pull request by nitincchauhan**
could not fix TestConcurrentBag, I am not sure what to pass as parameters there

### CODE REVIEW

File: src/main/java/com/zaxxer/hikari/pool/HikariPool.java [hunk 1]
```diff
@@ -494,9 +494,9 @@ private boolean addConnection()
          String password = config.getPassword();
 
          connection = (username == null) ? dataSource.getConnection() : dataSource.getConnection(username, password);
-         poolElf.setupConnection(connection, connectionTimeout);
+         PoolBagEntry pbe = poolElf.createBagEntry(connection, connectionTimeout, this);
 
-         connectionBag.add(new PoolBagEntry(connection, this));
+         connectionBag.add(pbe);
```
Comment (brettwooldridge) [inline src/main/java/com/zaxxer/hikari/pool/HikariPool.java]: This is *less* readable to me.  On the right I can clearly see that the connection is being setup.  With this change, that action becomes a side-effect of creating a bag entry, and is non-obvious to the reader.

File: src/main/java/com/zaxxer/hikari/pool/PoolBagEntry.java [hunk 2]
```diff
@@ -68,18 +68,22 @@
       DATE_FORMAT = new SimpleDateFormat("MMM dd, HH:mm:ss.SSS");
    }
 
-   public PoolBagEntry(final Connection connection, final HikariPool pool)
+   public PoolBagEntry(final Connection connection, final HikariPool pool, final boolean isReadOnly, final String catalog, final boolean isAutoCommit, final int networkTimeout, final int transactionIsolation)
```
Comment (brettwooldridge) [inline src/main/java/com/zaxxer/hikari/pool/PoolBagEntry.java]: This is getting close to violating standards on the [number of parameters](http://checkstyle.sourceforge.net/config_sizes.html#ParameterNumber).

File: src/main/java/com/zaxxer/hikari/pool/PoolElf.java [hunk 3]
```diff
@@ -171,8 +171,9 @@ else if (jdbcUrl != null && dataSource == null) {
     * @param connection a Connection
     * @param connectionTimeout the connection timeout
     * @throws SQLException thrown from driver
+    * @return PoolBagEntry
     */
-   void setupConnection(final Connection connection, final long connectionTimeout) throws SQLException
+   PoolBagEntry createBagEntry(final Connection connection, final long connectionTimeout, HikariPool pool) throws SQLException
```
Comment (brettwooldridge) [inline src/main/java/com/zaxxer/hikari/pool/PoolElf.java]: Finally, now PoolElf has turned into a factory for PoolBagEntry objects?

File: src/main/java/com/zaxxer/hikari/pool/HikariPool.java (hunk 1 above)
Comment (nitincchauhan) [inline src/main/java/com/zaxxer/hikari/pool/HikariPool.java]: true. but I am sure it is not that difficult to understand either having pbe defined right above the line.
this change removes juggling of method calls while creating bag entry with one one less method

File: src/main/java/com/zaxxer/hikari/pool/PoolElf.java [hunk 4]
```diff
@@ -281,15 +284,6 @@ void resetConnectionState(final PoolBagEntry poolEntry) throws SQLException
...
 
-   void resetPoolEntry(final PoolBagEntry poolEntry)
-   {
-      poolEntry.setReadOnly(isReadOnly);
-      poolEntry.setCatalog(catalog);
-      poolEntry.setAutoCommit(isAutoCommit);
-      poolEntry.setNetworkTimeout(networkTimeout);
-      poolEntry.setTransactionIsolation(transactionIsolation);
-   }
-
```
Comment (brettwooldridge) [inline src/main/java/com/zaxxer/hikari/pool/PoolElf.java]: All that for the purpose of removing ``resetPoolEntry()``?

This is a lot of change, that makes the code harder to follow, move responsibilities to unobvious places, and is ultimately in such a low traffic path  it likely gets executing less than once per minute.

Comment (nitincchauhan) [inline src/main/java/com/zaxxer/hikari/pool/PoolElf.java]: To me it is much simpler now, earlier code was HARD to follow having call to resetPoolEntry() from constructor and that method was calling methods on self to set each member.

File: src/main/java/com/zaxxer/hikari/pool/HikariPool.java (hunk 1 above)
Comment (brettwooldridge) [inline src/main/java/com/zaxxer/hikari/pool/HikariPool.java]: What I mean is, on the left I clearly see a method ``setupConnection()``, which clearly indicates what it does.

With the ``createBagEntry()`` call, I have no indication that it might execute SQL or otherwise change the connection state.  It "appears" to simply create a ``PoolBagEntry``.

Lastly, passing ``connectionTimeout`` into ``createBagEntry()`` make it appear that it is a part of the bag entry *state*.  However, it is only used during connection setup, which is crystal clear in the code on the left.

File: src/main/java/com/zaxxer/hikari/pool/PoolElf.java (hunk 4 above)
Comment (brettwooldridge) [inline src/main/java/com/zaxxer/hikari/pool/PoolElf.java]: I will grant that ``resetPoolEntry()`` is not clear.  When naming the method, it was originally was called from more than one place.

I would suggest reverting all of the changes above, and renaming this method to something like ``initializeDefaultState()`` or ``configureStateFromDefaults()``, ...

The ``PoolElf`` holds all of the pool defaults derived from the ``HikariConfig`` and after determining supported APIs, etc.  That is why it is responsible for ``resetConnectionState(PoolBagEntry)`` as well as ``resetPoolEntry(PoolBagEntry)``.  While the former is clearly named, the later is clearly not.

Comment (nitincchauhan) [inline src/main/java/com/zaxxer/hikari/pool/PoolElf.java]: close this please. sorry for your time.

Comment (nitincchauhan) [inline src/main/java/com/zaxxer/hikari/pool/PoolElf.java]: BTW renaming method is not going to solve mess of back and forth method calls.
Thanks
